### DISCUSSION TITLE
Project template for xplat FuncUI Avalonia, Elmish style?

### DISCUSSION THREAD

**Discussion by bentayloruk** [fsprojects/Avalonia.FuncUI#279] [Q&A] [answered]
Hello. Are there any examples or a project template, that will set me up with FuncUI for an xplat Elmish mobile app (iOS and Android)? I think I'm looking for something that is a cross between the `avalonia.xplat` template (xplat Avalonia but not Elmish) and one of the Elmish examples [here](https://github.com/fsprojects/Avalonia.FuncUI/tree/master/src/Examples/Elmish%20Examples) (Elmish but not setup for iOS, Android etc). Any pointers would be much appreciated!

**Comment by JordanMarr** [comment]
I would start with the Avalonia template and then manually add the FuncUI packages and copy the relevant lines in Program.fs from the examples. It’s more straightforward than you might think.

**Reply by bentayloruk** [reply]
Great, I shall give it a go! I thought I would ask first as you can sometimes spend ages trying to figure this out when on a new framework (e.g. there is some magic bit you have to flip somewhere etc). Thanks for the reply.

**Comment by AngelMunoz** [comment, marked as answer]
Take alook at this PR
https://github.com/AvaloniaUI/avalonia-dotnet-templates/pull/102

It uses the component version but it should be relatively easy to change it to the most up to date version and may give you some pointers to get started

**Reply by JaggerJo** [reply]
Here is a minimal sample iOS app. 

Probably just a few lines of code ore required to (also) target android.

You might want to break it out into different projects 

- shared (targets net6.0)
- ios (targets net6.0-ios)
- android (targets net6.0-android)

**Reply by bentayloruk** [reply]
Thanks @AngelMunoz! I have a look at this later today. Thanks for the additional guidance too @JaggerJo.

**Reply by AngelMunoz** [reply]
Hey by the way I just updated the templates I had here: https://github.com/AngelMunoz/AvaloniaTest 

That is the xplat template I based myself to make that closed PR above
It might be worth making a pull request to integrate one of the versions of this in the FuncUI repo
I'm not sure if the https://www.nuget.org/packages/JaggerJo.Avalonia.FuncUI.Templates nuget is up to date with the current versions + templates

cc @JordanMarr @JaggerJo @sleepyfran
